### PR TITLE
Inga/bettingfix

### PR DESCRIPTION
**Pull request by ingawei**
Cleaning up betting UI

### CODE REVIEW

File: web/components/contract/contract-details.tsx [hunk 1]
```diff
@@ -120,74 +127,157 @@ export function ContractDetails(props: {
...
+        </div>
+      )}
+      {/* {user && (
+        <>
+          <Row className="hidden items-center gap-1 md:inline-flex">
+            <DatabaseIcon className="h-5 w-5" />
+            <div className="whitespace-nowrap">{volumeLabel}</div>
+          </Row>
+        </>
+      )} */}
```
Comment (jahooma) [inline web/components/contract/contract-details.tsx:220]: nit: it's almost always not a great idea to commit commented out code.
Better to just remove. Someone could always dig it up in the git history.

File: web/components/contract/contract-details.tsx [hunk 2]
```diff
@@ -120,74 +127,157 @@ export function ContractDetails(props: {
...
+                {!resolvedDate && closeTime && user && (
+                  <Row>
+                    <div>Closes&nbsp;</div>
+                    <EditableCloseDate
+                      closeTime={closeTime}
+                      contract={contract}
+                      isCreator={isCreator ?? false}
+                    />
+                  </Row>
+                )}
```
Comment (jahooma) [inline web/components/contract/contract-details.tsx:187]: Decent amount of nesting here, and this is logically related. A good candidate for extracting out into another component! E.g. `CloseOrResolveTime`

File: web/components/contract/contract-details.tsx [hunk 3]
```diff
@@ -120,74 +127,157 @@ export function ContractDetails(props: {
...
-                size={'xs'}
-                color={'gray-white'}
+            {user && (
+              <button
+                className="text-greyscale-4 hover:text-greyscale-3"
                 onClick={() => setOpen(!open)}
               >
-                <PencilIcon className="mb-0.5 mr-0.5 inline h-4 w-4 shrink-0" />
-              </Button>
+                <PlusCircleIcon className="mb-0.5 mr-0.5 inline h-4 w-4 shrink-0" />
```
Comment (jahooma) [inline web/components/contract/contract-details.tsx:279]: Thanks, this icon is wayyy better than the pencil here.

File: web/components/contract/contract-details.tsx [hunk 4]
```diff
@@ -280,12 +332,12 @@ export function ExtraMobileContractDetails(props: {
         !resolvedDate &&
         closeTime && (
           <Col className={'items-center text-sm text-gray-500'}>
+            <Row className={'text-gray-400'}>Closes&nbsp;</Row>
```
Comment (jahooma) [inline web/components/contract/contract-details.tsx:335]: Ditto, substitute `{' '}` for the non-breaking space.

File: web/components/follow-button.tsx [hunk 5]
```diff
@@ -69,3 +73,36 @@ export function UserFollowButton(props: { userId: string; small?: boolean }) {
     />
   )
 }
+
+export function MiniUserFollowButton(props: { userId: string }) {
+  const { userId } = props
+  const currentUser = useUser()
+  const following = useFollows(currentUser?.id)
+  const isFollowing = following?.includes(userId)
+  const user = useUser()
```
Comment (jahooma) [inline web/components/follow-button.tsx:82]: You're using `useUser` twice. I think the second one should be `useUserById(userId)`

Comment (jahooma) [reply web/components/follow-button.tsx:82]: Or, you might not need it at all.

**Review by jahooma — APPROVED** [review]
Nice change, Inga! Getting the market page layout right is really important, and I like the push to toward simplicity, explaining the close date, and the more-compact follow button.

Some notes on the PR/code:
- It's a good idea to include screenshots in the PR description (even if slightly out of date haha). Makes it easier for us all to scrutinize the changes.
- Commenting out code is fine for drafts, but not for code that is pushed. Make sure to remove those before merging.
- Code looks really nice in general!

On the design:
- Like Stephen was saying in standup, I don't think we want to hide the like and watch buttons within the '...' menu. Seems like they would fit fine right next to the share button in the header?
- Did you mean for there to be a second share button below the graph?
- What do you think about showing more groups, say, up to 3?
- The follow button for creators is cool, but I'm sure how clear it is. What if you get one of those toast notifications (popping down from the top) that says you have followed them when clicked? For an example, see `like-market-button.tsx`.
- Is there no way to unfollow the creator from this page? Maybe that's fine?

File: web/components/contract/contract-details.tsx [hunk 6]
```diff
@@ -120,74 +127,157 @@ export function ContractDetails(props: {
...
+                        <div>resolved&nbsp;</div>
+                        <b>{resolvedDate}</b>
+                      </Row>
+                    </DateTimeTooltip>
+                  </>
+                ) : null}
+
+                {!resolvedDate && closeTime && user && (
+                  <Row>
+                    <div>Closes&nbsp;</div>
```
[the remark this answers is not in the document]
Comment (ingawei) [reply web/components/contract/contract-details.tsx:180]: For some reason it just replaces it with a regular space and then doesn't actually translate? Also sorry this code is such a mess atm, I assumed no one would look at it until I assigned it to someone, lmk if I just shouldn't create the request in the future until it's review ready

**Comment by ingawei** [comment]
> * Like Stephen was saying in standup, I don't think we want to hide the like and watch buttons within the '...' menu. Seems like they would fit fine right next to the share button in the header?

Yeah maybe I can put it in web view, but it's definitely too cluttered in mobile

File: web/components/contract/contract-details.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (jahooma) [reply web/components/contract/contract-details.tsx:180]: Np. Sorry for barging in with a review! You could always mark it as a draft PR. Though I tend to think it's good to get feedback as soon as possible.

Hmm, maybe just add some margin right then instead haha.

[the remark this answers is not in the document]
Comment (ingawei) [reply web/components/contract/contract-details.tsx:180]: Oh no I super appreciate the feedback! Your comments are always really helpful I'm just a little embarrassed at the state of this PR at the moment loll. Ehh is that less bad? The text size changes from web to mobile, I feel like conditional margins on this is kind of gross

File: web/components/contract/contract-details.tsx [hunk 7]
```diff
@@ -104,6 +106,11 @@ export function AvatarDetails(props: {
   )
 }
 
+export function getIsMobile() {
```
[the remark this answers is not in the document]
Comment (ingawei) [reply web/components/contract/contract-details.tsx:109]: Done

File: web/components/contract/contract-details.tsx (hunk 1 above)
Comment (ingawei) [reply web/components/contract/contract-details.tsx:220]: Done

File: web/components/contract/contract-details.tsx (hunk 2 above)
Comment (ingawei) [reply web/components/contract/contract-details.tsx]: Done

File: web/components/contract/contract-details.tsx (hunk 3 above)
Comment (ingawei) [reply web/components/contract/contract-details.tsx:279]: Thank you! :)

**Pull request merged**